### ISSUE TITLE
Replace all hard-coded pluralize("crop") and similar with Crop.model_name.human()

### ISSUE THREAD

**Issue by maco** [Growstuff/growstuff#919]
We hard-code strings for each model and then tell the system to pluralize them. This means these model names aren't automatically translated. We should instead rely on `.model_name.human(count)` which will pull from config/locale/en.yml (or es.yml or jp.yml etc) automatically, depending on which language of the site visitors are viewing.

## To fix this:

1. Search the whole codebase for `pluralize(`
2. If it is *not* using one of our model names (from `app/models/`) in quotes as the second thing in the parentheses, just skip over it. That's a more advanced case.
2. When you find it, check to see if the model being used is defined in the `activerecord` section [of the English translation file](https://github.com/Growstuff/growstuff/blob/dev/config/locales/en.yml)
3. If it isn't, add it to that file, just like the others (specifying `one` with the singular spelling and `other` with the plural)
4. Replace the `pluralize(count, "model")` call with `Model.model_name.human(:count => count)`

## Example:

The Activity section on the Member page has:

```
link_to pluralize(member.posts.size, "post"), posts_by_author_path(:author => member)
```

It should be:

```
link_to Post.model_name.human(:count => member.posts.size), posts_by_author_path(:author => member)
```

**Comment by lucasnogueira** [comment]
Hello. I was just checking this task and I got a question. Changing the methods will make the actual number of objects disappear. For exemple:

`pluralize(2, 'foo')` equals `'2 foos'` while `Foo.model_name.human(:count => 2)` will result in just `'foos'`.

Is this the expected behaviour?

**Comment by maco** [comment]
Oops, my instructions are not quite right, then. Still put in the number!

**Comment by lucasnogueira** [comment]
So I suppose we should keep the `pluralize` but  replace the hard coded strings with the `Model.model_name.human` right?

**Comment by maco** [comment]
Possibly something like this:

`link_to "#{member.posts.size} #{Post.model_name.human(:count => member.posts.size)}", posts_by_author_path(:author => member)`

So we're making a string for the link text, and that string is made up of:

- the number
- the correctly pluralized noun in whatever language the visitor is using

**Comment by maco** [comment]
(Does that make sense?)

**Comment by lucasnogueira** [comment]
Yeah it makes! But is there any problem with the `pluralize`? This option came to my mind but it feels a little bit repetitive. `pluralize(member.posts.size, Post.model_name.human)` would have the same result and would take care of the hardcoded string (that I believe is the problem here).

**Comment by maco** [comment]
If I'm understanding the docs correctly, using `pluralize()` you have to
pass in a locale parameter for languages other than English (otherwise
it'll pluralize the word using English rules). If we pass a count to
`Model.model_name.human` then the target language's pluralization rules
automatically take over without us having to read the locale from the
environment and pass it in. (But I could be wrong. I haven't worked on a
localized project before.)

**Comment by pozorvlak** [comment]
My reading of the docs is also that [`pluralize` needs an explicit `locale` parameter](http://apidock.com/rails/ActiveSupport/Inflector/pluralize) but [`model_name.human` picks one up out of the ether](http://api.rubyonrails.org/classes/ActiveModel/Name.html#method-i-human), but it would be good to add some tests for correct pluralization at the same time as we make this change. AIUI Japanese pluralization rules are different from English ones, so this shouldn't involve adding a new language :-)

I also note that `pluralize` [incorrectly pluralizes "CamelOctopus"](http://apidock.com/rails/ActiveSupport/Inflector/pluralize) as "CamelOctopi", when as any fule kno it should be "[CamelOctopuses](https://www.youtube.com/watch?v=wFyY2mK8pxk)"

**Comment by pozorvlak** [comment]
It looks like either approach would be a bit repetitive. In that case the Right Thing would IMHO be to extract out a helper along the lines of

```ruby
# Untested, probably won't work
def count_text(collection)
    "#{collection.size} #{collection.class.model_name.human(count: collection.size)}"
end
```
or
```ruby
# Untested, certainly won't work
def count_text(collection)
    pluralize(collection.size, collection.class.model_name.human, locale: get_locale_somehow())
end
```

**Comment by maco** [comment]
Japanese pluralization rules change the pronunciation but not how it's
written in kanji.

1 [person character] is pronounced hitori
2 [person character] is pronounced futari

Unless it's written phonetically (which I doubt pluralization would handle,
frankly - Japanese is way too heavy on homophones), the spelling of the
noun doesn't change.

Italian has more interesting pluralization...

**Comment by lucasnogueira** [comment]
Got it, just wanted to be 101% sure :)

Also about @pozorvlak idea, `collection.first.class.model_name.human(count: collection.size)` works so I am going for it if there is no problem.

**Comment by cesy** [comment]
Yeah, sending the count to the translation is the right way to do this.
Once I'm back from holiday, I can pull up links to examples in Arabic and
Russian which have interesting pluralisation. We've recently worked through
similar issues on AO3.

**Comment by pozorvlak** [comment]
@cesy: yeah, I'm trying to learn Russian at the moment and the pluralization rules are... challenging.

**Comment by pozorvlak** [comment]
Fixed by #980, AFAICT.
